crypto/src/hash/sha3/mod.rs

```diff
@@ -67,27 +71,45 @@ impl Sha3Hasher {
         a.iter().zip(b).map(|(a, b)| a ^ b).collect()
     }
 }
-impl<F: IsField> IsCryptoHash<F> for Sha3Hasher {
-    fn hash_one(&self, input: &FieldElement<F>) -> FieldElement<F>
-    where
-        FieldElement<F>: ByteConversion,
-    {
+impl<M: IsModulus<UnsignedInteger<N>> + Clone, const N: usize>
+    IsCryptoHash<MontgomeryBackendPrimeField<M, N>> for Sha3Hasher
+{
+    fn hash_one(
+        &self,
+        input: &FieldElement<MontgomeryBackendPrimeField<M, N>>,
+    ) -> FieldElement<MontgomeryBackendPrimeField<M, N>> {
         let mut hasher = Sha3_256::new();
         hasher.update(input.to_bytes_be());
         let mut result_hash = [0_u8; 32];
         result_hash.copy_from_slice(&hasher.finalize());
-        FieldElement::<F>::from_bytes_le(&result_hash).unwrap()
+        FieldElement::from_bytes_le(&result_hash).unwrap()
     }
 
-    fn hash_two(&self, left: &FieldElement<F>, right: &FieldElement<F>) -> FieldElement<F>
-    where
-        FieldElement<F>: ByteConversion,
-    {
+    fn hash_two(
+        &self,
+        left: &FieldElement<MontgomeryBackendPrimeField<M, N>>,
+        right: &FieldElement<MontgomeryBackendPrimeField<M, N>>,
+    ) -> FieldElement<MontgomeryBackendPrimeField<M, N>> {
         let mut hasher = Sha3_256::new();
         hasher.update(left.to_bytes_be());
         hasher.update(right.to_bytes_be());
         let mut result_hash = [0_u8; 32];
         result_hash.copy_from_slice(&hasher.finalize());
-        FieldElement::<F>::from_bytes_le(&result_hash).unwrap()
+        FieldElement::from_bytes_le(&result_hash).unwrap()
+    }
```

Comment:
Why don't we just either remove these and always call `hash_many` or turn them into wrappers for `hash_many`?